core/views.py

```diff
@@ -446,6 +448,44 @@ def get_object(self):
 
 profile_edit = ProfileEditView.as_view()
 
+class ConnectedUserCircleEditView(UpdateView):
+    model = models.User  # or CircleMembership, depending on your data model
+    form_class = forms.ConnectedUserCircleForm
+    template_name = 'core/edit_connected_user_circles.html'
+    success_url = reverse_lazy('connection_list')  # Adjust as necessary
+
+    def get_object(self):
+        # Ensure the user being edited is connected to the request.user
+        target_user = get_object_or_404(models.User, pk=self.kwargs['pk'])
+        if not self.request.user.is_connected_with(target_user):
+            raise PermissionDenied
+        return target_user
+
+    def form_valid(self, form):
+        circles = form.cleaned_data['circles']
+        target_user = self.get_object()
+        selected_circles = form.cleaned_data['circles']
+        non_selected = models.Circle.objects.exclude(
+            id__in=[c.id for c in selected_circles])
+
+        models.CircleMembership.objects.filter(
+            circle__in=non_selected,
+            connection__other_user=target_user
+        ).delete()
+
+        for circle in circles:
+            models.Circle.objects.get_or_create(
+                name=circle.name, owner=self.request.user)
+        return super().form_valid(form)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['circles'] = self.request.user.circles
+        kwargs['user'] = self.get_object()  # Pass the target user to the form
+        return kwargs
+
+e_c = ConnectedUserCircleEditView.as_view()
```

Comment:
`e_c` should definitely be a more descriptive name. Maybe `edit_connection_circles` or something.